erpnext/accounts/doctype/payment_manager/payment_manager.py

```diff
@@ -0,0 +1,429 @@
+		return payment
+
+	def create_journal_entry(self, pe):
+		je = frappe.new_doc("Journal Entry")
+		je.is_opening = "No"
+		je.voucher_type = "Bank Entry"
+		je.cheque_no = pe.description
+		je.cheque_date = pe.transaction_date
+		je.remark = pe.description
+		je.posting_date = pe.transaction_date
+		if (pe.amount < 0):
+			je.append("accounts", {"account": pe.party, "debit_in_account_currency": abs(pe.amount)})
+			je.append("accounts", {"account": self.bank_account, "credit_in_account_currency": abs(pe.amount)})
+		else:
+			je.append("accounts", {"account": pe.party, "credit_in_account_currency": pe.amount})
+			je.append("accounts", {"account": self.bank_account, "debit_in_account_currency": pe.amount})
+		je.save()
+		return je
+
+	def update_payment_entry(self, payment):
+		lst = []
+		invoices = payment.invoices.strip().split(',')
+		if (len(invoices) == 0): return
+		amount = float(abs(payment.amount))
+		for invoice_entry in invoices:
+			if (not invoice_entry.strip()): continue
+			invs = invoice_entry.split('|')
+			invoice_type, invoice = invs[0], invs[1]
+			outstanding_amount = frappe.get_value(invoice_type, invoice, 'outstanding_amount')
+
+			lst.append(frappe._dict({
+				'voucher_type': payment.reference_type,
+				'voucher_no' : payment.reference_name,
+				'against_voucher_type' : invoice_type,
+				'against_voucher'  : invoice,
+				'account' : payment.account,
+				'party_type': payment.party_type,
+				'party': frappe.get_value("Payment Entry", payment.reference_name, "party"),
+				'unadjusted_amount' : float(amount),
+				'allocated_amount' : min(outstanding_amount, amount)
+			}))
+			amount -= outstanding_amount
+		if lst:
+			from erpnext.accounts.utils import reconcile_against_document
+			try:
+				reconcile_against_document(lst)
+			except:
+				frappe.throw("Exception occurred while reconciling {0}".format(payment.reference_name))
+
+	def submit_payment_entries(self):
+		for payment in self.new_transaction_items:
+			if payment.reference_name is None: continue
+			doc = frappe.get_doc(payment.reference_type, payment.reference_name)
+			if doc.docstatus == 1:
+				if (payment.reference_type == "Journal Entry"): continue
+				if doc.unallocated_amount == 0: continue
+				print("Reconciling payment {0}".format(payment.reference_name))
+				self.update_payment_entry(payment)
+			else:
+				print("Submitting payment {0}".format(payment.reference_name))
+				if (payment.reference_type == "Payment Entry"):
+					if (payment.payment_reference):
+						doc.reference_no = payment.payment_reference
+					doc.mode_of_payment = payment.mode_of_payment
+				doc.save()
+				doc.submit()
+		self.move_reconciled_entries()
+		self.populate_matching_invoices()
+
+	def move_reconciled_entries(self):
+		idx = 0
+		while idx < len(self.new_transaction_items):
+			entry = self.new_transaction_items[idx]
+			print("Checking transaction {0}: {2} in {1} entries".format(idx, len(self.new_transaction_items), entry.description))
```

Comment:
remove print

Note: Please remove in all places.

Reply:
Sure. Btw, what is the standard way to log anything of importance?